src/app/archive/create_schema.sql

```diff
@@ -142,52 +142,49 @@ CREATE INDEX idx_blocks_creator_id ON blocks(creator_id);
-   they are included here to allow Rosetta account queries to consume
-    fewer Postgresql resources
-   TODO: nonce column is NULLable until we can establish valid nonces for all rows
+CREATE TABLE account_ids
+( id                 serial  PRIMARY KEY
+, public_key_id      int     NOT NULL     REFERENCES public_keys(id) ON DELETE CASCADE
+, token              text    NOT NULL
+, token_owner        int                  REFERENCES account_ids(id)
+);
+
+/* accounts accessed in a block, representing the account
+   state after all transactions in the block have been executed
 */
 
-CREATE TABLE balances
-( id                           serial PRIMARY KEY
-, public_key_id                int    NOT NULL REFERENCES public_keys(id)
-, balance                      bigint NOT NULL
-, block_id                     int    NOT NULL REFERENCES blocks(id) ON DELETE CASCADE
-, block_height                 int    NOT NULL
-, block_sequence_no            int    NOT NULL
-, block_secondary_sequence_no  int    NOT NULL
-, nonce                        bigint
-, UNIQUE (public_key_id,balance,block_id,block_height,block_sequence_no,block_secondary_sequence_no)
+CREATE TABLE accounts_accessed
+( ledger_index            int     NOT NULL
+, block_id                int     NOT NULL  REFERENCES blocks(id)
+, account_id              int     NOT NULL  REFERENCES account_ids(id)
+, token_symbol            text    NOT NULL
+, balance                 bigint  NOT NULL
+, nonce                   bigint  NOT NULL
+, receipt_chain_hash      text    NOT NULL
+, delegate                int               REFERENCES public_keys(id)
+, voting_for              text    NOT NULL
+, timing                  int               REFERENCES timing_info(id)
+, permissions             int     NOT NULL  REFERENCES zkapp_permissions(id)
+, zkapp                   int               REFERENCES zkapp_accounts(id)
+, PRIMARY KEY (block_id,account_id)
 );
 
-CREATE INDEX idx_balances_id ON balances(id);
-CREATE INDEX idx_balances_public_key_id ON balances(public_key_id);
-CREATE INDEX idx_balances_height_seq_nos ON balances(block_height,block_sequence_no,block_secondary_sequence_no);
+CREATE INDEX idx_accounts_accessed_block_id ON accounts_accessed(block_id);
+CREATE INDEX idx_accounts_accessed_block_account_id ON accounts_accessed(account_id);
 
 CREATE TABLE blocks_user_commands
 ( block_id        int NOT NULL REFERENCES blocks(id) ON DELETE CASCADE
 , user_command_id int NOT NULL REFERENCES user_commands(id) ON DELETE CASCADE
 , sequence_no     int NOT NULL
 , status          user_command_status NOT NULL
 , failure_reason  text
-, fee_payer_account_creation_fee_paid bigint
-, receiver_account_creation_fee_paid bigint
-, created_token     text
-, fee_payer_balance int NOT NULL REFERENCES balances(id) ON DELETE CASCADE
-, source_balance    int          REFERENCES balances(id) ON DELETE CASCADE
-, receiver_balance  int          REFERENCES balances(id) ON DELETE CASCADE
+, created_token   text
```

Comment:
We can't create custom tokens using old transactions, can we? afaik this table is only for payments and delegations. Did you mean to add this in `blocks_zkapp_commands`?

Reply:
the column is deleted in the PR stacked above this one, it will disappear when that's merged